app/pages/learn/01_tutorial/01_your-first-java-app/02_building-with-intellij-idea.md

```diff
@@ -0,0 +1,221 @@
+
+IntelliJ IDEA supports multiple testing libraries, including [JUnit5](https://junit.org/junit5/), which is the [most used testing library for Java developers](https://www.jetbrains.com/lp/devecosystem-2023/java/#java_unittesting).
+
+If JUnit5 is not part of your project yet, IntelliJ IDEA will note “JUnit5 library not found in the module”. Click **Fix** to have IntelliJ IDEA fix this for you.
+
+[![Create test - Fix](/assets/images/intellij-idea/create-test-fix.png)](/assets/images/intellij-idea/create-test-fix.png)
+
+Note that JUnit5 dependencies are added to the pom.xml.
+
+[![JUnit5 dependencies](/assets/images/intellij-idea/junit5-dependencies.png)](/assets/images/intellij-idea/junit5-dependencies.png)
+
+Go back to the test file to add tests. We can let IntelliJ IDEA help us generate our test for us. In the test class, we can use **Generate** (**⌘N** on macOS or **Alt+Insert** on Windows/Linux) and select **Test Method** to add a test. Give the test a name that explains the intended behavior, and add the relevant test code.
+
+In our test class, we can select **Run All Tests** (**⌃⇧R** on macOS or **Ctrl+Shift+F10** on Windows/Linux).
+While it’s nice to see our tests pass, we also want to make sure that they fail when something is wrong. Make some changes to your code that will make your tests fail, and run them again to see them fail. Revert those changes to see them pass again.
+
+We can view the code and tests side by side. To do so, click on the tab with the test file, and select **Move and Split right**.
+
+## Debugging
+We might want to see how our code runs, either to help us understand how it works and/or when we need to fix a bug. We can run our code through the [debugger](https://www.jetbrains.com/help/idea/debugging-code.html) to see the state of our variables at different times, and the call stack - or the order in which methods are called when the program executes. To do so, we must first add a [breakpoint](https://www.jetbrains.com/help/idea/using-breakpoints.html) to the code.
+
+To add a breakpoint, click the gutter at the line of code where you want execution to stop. Alternatively, place the caret at the line and press **⌃F8** (on macOS) or **Ctrl+F8** (on Windows/Linux). We can run our test or application using the **Debug** option.
+
+Execution will stop at the breakpoint, so we can investigate the state of our application. We can see current values of variables and objects. We can evaluate an expression, to see its current value and look at more details. We can even change the expressions to evaluate different results. We can continue execution by either stepping into a method to see what happens inside a called method (using the shortcut **F7**) or stepping over a line to go to the next line even if a method is called (using the shortcut **F8**), depending on what we’re interested in. Finally, we can resume the program to finish the execution of the test.
```

Comment:
There's already an [article about debugging](https://dev.java/learn/debugging/) which also links to the IntelliJ documentation.
I don't think there's any point in explaining how debugging works here (other than maybe linking to the debugging article).

Reply:
Agreed, it might be good to link to that article here.

Reply:
Added a link.